neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -166,54 +176,78 @@ async def build_tool_call_result(
         }
         # Construct actual tool call result:
         if result_structure is not None:
-            call_result["result"]["structuredContent"] = result_structure
+            call_result["result"]["structuredContent"] = {}
+            call_result["result"]["structuredContent"]["result"] = result_structure
             # For backward compatibility, also add text version of structure:
             structure_str: str = f"```json\n{json.dumps(result_structure, indent=2)}\n```"
             result_text = result_text + structure_str
+        if result_sly_data is not None:
+            call_result["result"]["structuredContent"]["sly_data"] = result_sly_data
         call_result["result"]["content"][0]["text"] = RequestsUtil.safe_message(result_text)
         return call_result
 
-    async def _get_tool_description(self, agent_name: str, metadata: Dict[str, Any]) -> Dict[str, Any]:
-        service_provider: AsyncAgentServiceProvider = self.agent_policy.allow(agent_name)
-        if service_provider is None:
-            return None
-        service: AsyncAgentService = service_provider.get_service()
-        function_dict: Dict[str, Any] = await service.function({}, metadata)
-        tool_description: str = function_dict.get("function", {}).get("description", "")
+    async def _get_tool_description(self, agent_info_dict: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Get MCP tool description for given agent.
+        :param agent_info_dict: dictionary with agent information;
+        :return: MCP tool description dictionary
+        """
         return {
-            "name": agent_name,
-            "description": tool_description,
+            "name": agent_info_dict.get("agent_name", ""),
+            "description": agent_info_dict.get("description", ""),
+            "tags": agent_info_dict.get("tags", ""),
             "inputSchema": {
                 "type": "object",
                 "properties": {
                     "input": {
                         "type": "string",
                         "description": "text input for chat request"
+                    },
+                    "chat_context": {
+                        "type": "string",
+                        "description": "optional chat context"
+                    },
+                    "sly_data": {
+                        "type": "object",
+                        "description": "optional arbitrary JSON dictionary",
+                        "additionalProperties": True
                     }
```

Comment:
Extend published MCP tools schema with additional parameters.

Reply:
More of the same schema errors.
Why do we need a new schema for MCP?
Why can't this follow what is needed for ChatRequest in neuro_san/api/grpc/agent.proto?